### PR TITLE
Integeration/setting profile endpoints

### PR DESCRIPTION
**Pull request by ejaz-hussain-1050**
Settings integration with backend

### CODE REVIEW

File: backend/functions/src/api/resources/user-profile/user-profile.model.ts [hunk 1]
```diff
@@ -5,4 +5,7 @@ export interface UserProfile {
   gender?: 'male' | 'female' | '';
   name?: string;
   bio?: string;
+  createdAt?: Date;
+  updatedAt?: Date;
+  isDeleted?: boolean;
```
Comment (alinauman02) [inline backend/functions/src/api/resources/user-profile/user-profile.model.ts:10]: We will add these fields in every model. Should we keep copy pasting them like this, or is there some way to make this reusable?

File: frontend/src/apis/create-api.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
+import { UserProfile } from 'models';
+
+enum ApiTag {
+  PROFILE = 'Profile',
+}
+export interface Post {
```
Comment (alinauman02) [inline frontend/src/apis/create-api.ts]: Why is this named `Post`? Isn't `Post` a specific model in the case of this app?

File: frontend/src/components/Login/Signup.tsx [hunk 3]
```diff
@@ -31,15 +33,19 @@ export function Signup() {
     setUserCredentials(currentuserCredentials => ({ ...currentuserCredentials, [name]: value }));
   };
 
-  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+  const OnSignUp = async (e: React.FormEvent<HTMLFormElement>) => {
```
Comment (alinauman02) [inline frontend/src/components/Login/Signup.tsx]: Using PascalCase? camelCase is sad :cry:

File: frontend/src/apis/create-api.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  id: string;
+  profile: UserProfile;
+}
+
+// Define a service using a base URL and expected endpoints
+export const api = createApi({
+  baseQuery: fetchBaseQuery({ baseUrl: 'http://127.0.0.1:3002/aurora-instagram-clone/us-central1/api' }),
```
Comment (alinauman02) [inline frontend/src/apis/create-api.ts]: You are also using the url string in the `auth.ts` file. In this file, extract this URL into a separate string variable and export it. Then reuse it in the `auth.ts` file.

File: frontend/src/components/Login/Signup.tsx [hunk 5]
```diff
@@ -31,15 +33,19 @@ export function Signup() {
...
     e.preventDefault();
-    try {
-      const res = await signUp(userCredentials.email, userCredentials.password);
-      if (res) navigate('/home');
-    } catch (err) {
-      if (err instanceof FirebaseError) {
-        setError(err.message);
-      }
+    const res = await signUpApi(
```
Comment (alinauman02) [inline frontend/src/components/Login/Signup.tsx]: Why did you remove the `try/catch` block?

Whenever you are dealing with a `Promise`, you need to handle the case where the `Promise` rejects with an error.

This means using `.catch` if you are using `.then`, or using a `try`/`catch` block in case of `async`/`await`.

Bring the block back.

File: frontend/src/components/Login/Signup.tsx [hunk 6]
```diff
@@ -31,15 +33,19 @@ export function Signup() {
...
+      userCredentials.name,
+      userCredentials.email,
+      userCredentials.password,
+      userCredentials.username
+    );
+    if (res.error) {
```
Comment (alinauman02) [inline frontend/src/components/Login/Signup.tsx]: If `res.error` is truthy, throw a new `Error`, with `res.error` as the `message`.

Also avoid unnecessary nesting by moving the `login` and `navigate` calls outside the `else` block. If you throw the `Error` above, you don't have to worry about these methods being called when `Promise` rejects.

File: frontend/src/components/settings/EditProfile.tsx [hunk 7]
```diff
@@ -69,7 +92,7 @@ export function EditProfile() {
             onInput={onChangeField}
             rows={3}
             maxLength={150}
-            value={profileInfo.bio}
+            value={profileInfo.bio ? profileInfo.bio : ''}
```
Comment (alinauman02) [inline frontend/src/components/settings/EditProfile.tsx]: In this case, you can use the [nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing).

You could also do this checking when setting the `profileInfo` state.

File: frontend/src/store/store.ts [hunk 8]
```diff
@@ -1,7 +1,13 @@
 import { configureStore } from '@reduxjs/toolkit';
+import { api } from 'apis/create-api';
+import { authSliceReducer } from './slices/auth-slice';
```
Comment (alinauman02) [inline frontend/src/store/store.ts]: Change the paths to use the barrel files

**Review by alinauman02 — CHANGES_REQUESTED** [review]
Did an initial review. Please address the comments.

**Review by alinauman02 — DISMISSED** [review]

File: frontend/src/config/firebase.ts [hunk 9]
```diff
@@ -14,4 +14,6 @@ const firebaseConfig = {
 // Initialize Firebase
 const app = initializeApp(firebaseConfig);
 const auth = getAuth(app);
+connectAuthEmulator(auth, 'http://localhost:3001');
```
Comment (ZunairaT) [inline frontend/src/config/firebase.ts]: store the url in an environment variable

**Review by ZunairaT — DISMISSED** [review]

**Review by alinauman02 — APPROVED** [review]

**Review by ZunairaT — APPROVED** [review]
